.github/pull_request_template.md

```diff
@@ -0,0 +1,50 @@
+# Description
+Mandatory, single line description of the pull request, written as though it was an order.
+
+Then, following a single blank line, a summary of the change and which issue is fixed, including 
+- [ ] what is the problem being solved
+- [ ] why this is the best approach
+- [ ] what are the shortcomings of this approach, if any
+It may include
+- [ ] background information, such as "Fixes #"
+- [ ] benchmark results
+- [ ] links to design documents
+- [ ] dependencies required for this change
+
+The summary may omit some of these details if this PR fixes a single ticket that includes these details. It is the reviewer's discretion. 
+
+## [Optional] Changes
+Multiple changes are not recommended, so this section should normally be omitted. Unforunately, they are sometimes unavoidable. If there are multiple logical changes, list them separately.
+
+1. `'foo'` is replaced by `'bar'`
+2. `'fizzbuzz'` is optimized for gas
+
+## Type of change
```

Comment:
I don't know if this section is necessary, if we continue to use conventional commits. But if we do so, I would ask for people to review their usage of the conventional commit nouns. For example, `fix` should be reserved for a _bug fix_,  and shouldn't used in a commit that fixes something that was broken in the ongoing PR.

Reply:
Does this seem like a good use case for labels?

They are more visible and something that we're partially used to using.

Reply:
I think we ought to end up using labels. @wunnaii do you have an opinion?

The reasons for including them in the PR template is:
1. it is one less "process" (you just check a box that's in front of you)
2. there are many labels. if some people use an `enhancement` label and some people use a `feature` label, that loses some value. It also adds some "mental overhead" -- I have to remember which label I should use for this type of PR.

Therefore, including a list of the specific ways we want to categorize PRs should make it easier to label things.

If we want to end up with labels, two options are:
A. expect developers to use the checkbox to select specific labels. They can then delete the checkbox.
B. Use the checkbox to automatically assign values with a GH action.

We could change the name & description of this section accordingly, if you prefer.
```suggestion
## Labels
```

Reply:
> A. expect developers to use the checkbox to select specific labels. They can then delete the checkbox.

If we're doing this I just think a list (instead of checkboxes) of labels that might be useful would be better. Otherwise, we'll probably end up with some people using the `checkbox`s to label their PR and other people using actual labels.

I'd rather enforce people just using `labels`.  We could have checkbox in the `checklist` for `appropriate labels selected`.

> B. Use the checkbox to automatically assign values with a GH action.

That'd be kinda slick but I feel like it might not be worth the effort?

Reply:
I prefer this idea. I'll be happy to remove this section. We can maintain a reference containing the meanings of the labels?

Reply:
2b48e5ca214f7b0d13957ca92a4add1bfa19b1df